File: Leetcode/1629/1629.cpp
class Solution {
public:
    char slowestKey(vector<int>& releaseTimes, string keysPressed) {
        constexpr int MAXN = 26;
        vector<vector<int> > A(MAXN);
        int n = (int)releaseTimes.size();
        for (int i = 0; i < n; ++i) {
            if (i == 0) {
                A[keysPressed[i] - 'a'].emplace_back(releaseTimes[0]);
            } else {
                A[keysPressed[i] - 'a'].emplace_back(releaseTimes[i]-releaseTimes[i-1]);
            }
        }
        vector<pair<char,int> > vec;
        for (int i = 0; i < MAXN; ++i) {
            if (!A[i].empty()) {
                vec.emplace_back(i+'a', *max_element(A[i].begin(), A[i].end()));
            }
        }
        return min_element(vec.begin(),vec.end(),[](const auto &l, const auto &r){
            if (l.second == r.second) {
                return l.first > r.first;
            }
            return l.second > r.second;
        })->first;
    }
};

File: Leetcode/1629/1629.go
func slowestKey(releaseTimes []int, keysPressed string) byte {
	const MAXN = 26
	var stats [MAXN]int
	for i := 0; i < len(keysPressed); i++ {
		if i == 0 {
			stats[keysPressed[i] - 'a'] = releaseTimes[i]
			continue
		}
		if stats[keysPressed[i] - 'a'] < releaseTimes[i] - releaseTimes[i-1] {
			stats[keysPressed[i] - 'a'] = releaseTimes[i] - releaseTimes[i-1]
		}
	}
	maxVal := 0
	var ans byte
	for i := MAXN - 1; i >= 0; i-- {
		if stats[i] > maxVal {
			maxVal = stats[i]
			ans = byte(i + 'a')
		}
	}
	return ans
}